gnomad/utils/constraint.py

```diff
@@ -0,0 +1,130 @@
+
+    if context_length == 3:
+        mid_index = 1
+    elif context_length == 7:
+        mid_index = 3
+    else:
+        raise ValueError(
+            f"The length of context should be either 3 or 7, instead of {context_length}."
+        )
+
+    transition_expr = hl.is_transition(t.ref, t.alt)
+    cpg_expr = (
+        (t.ref == "G") & (t.alt == "A") & (t.context[mid_index - 1 : mid_index] == "C")
+    ) | (
+        (t.ref == "C")
+        & (t.alt == "T")
+        & (t.context[mid_index + 1 : mid_index + 2] == "G")
+    )
+    if isinstance(t, hl.MatrixTable):
+        t = t.annotate_rows(transition=transition_expr, cpg=cpg_expr)
+    else:
+        t = t.annotate(transition=transition_expr, cpg=cpg_expr)
+    mutation_type_expr = (
+        hl.case()
+        .when(t.cpg, "CpG")
+        .when(t.transition, "non-CpG transition")
+        .default("transversion")
+    )
+    mutation_type_model_expr = hl.if_else(t.cpg, t.context, "non-CpG")
+    if isinstance(t, hl.MatrixTable):
+        return t.annotate_rows(
+            mutation_type=mutation_type_expr,
+            mutation_type_model=mutation_type_model_expr,
+        )
+    else:
+        return t.annotate(
+            mutation_type=mutation_type_expr,
+            mutation_type_model=mutation_type_model_expr,
+        )
+
+
+def trimer_from_heptamer(
+    t: Union[hl.MatrixTable, hl.Table]
+) -> Union[hl.MatrixTable, hl.Table]:
+    """
+    Trim heptamer context to create trimer context.
+
+    :param t: Input MatrixTable or Table with context annotation.
+    :return: MatrixTable or Table with trimer context annotated.
+    """
+    trimer_expr = hl.if_else(hl.len(t.context) == 7, t.context[2:5], t.context)
+    return (
+        t.annotate_rows(context=trimer_expr)
+        if isinstance(t, hl.MatrixTable)
+        else t.annotate(context=trimer_expr)
+    )
+
+
+def collapse_strand(
+    t: Union[hl.Table, hl.MatrixTable]
+) -> Union[hl.Table, hl.MatrixTable]:
+    """
+    Return the deduplicated context by collapsing DNA strands.
+
+    Function returns the reverse complement for 'ref, 'alt', and 'context' if the reference allele is either 'G' or 'T'.
+
+    The following annotations are added to the output Table:
+        - was_flipped - whether the 'ref, 'alt', and 'context' were flipped (reverse complement taken)
+
+    :param ht: Input Table.
+    :return: Table with deduplicated context annotation (ref, alt, context, was_flipped).
+    """
+    collapse_expr = {
+        "ref": hl.if_else(
+            ((t.ref == "G") | (t.ref == "T")), hl.reverse_complement(t.ref), t.ref
+        ),
+        "alt": hl.if_else(
+            ((t.ref == "G") | (t.ref == "T")), hl.reverse_complement(t.alt), t.alt
+        ),
+        "context": hl.if_else(
+            ((t.ref == "G") | (t.ref == "T")),
+            hl.reverse_complement(t.context),
+            t.context,
+        ),
+        "was_flipped": (t.ref == "G") | (t.ref == "T"),
+    }
```

Comment:
```suggestion
ref_g_or_t_expr = (t.ref == "G") | (t.ref == "T")

collapse_expr = {
        "ref": hl.if_else(
            (ref_g_or_t_expr), hl.reverse_complement(t.ref), t.ref
        ),
        "alt": hl.if_else(
            (ref_g_or_t_expr), hl.reverse_complement(t.alt), t.alt
        ),
        "context": hl.if_else(
            (ref_g_or_t_expr),
            hl.reverse_complement(t.context),
            t.context,
        ),
        "was_flipped": ref_g_or_t_expr,
    }
```

Reply:
just rewriting this so that the conditional statement doesn't need to be typed out as many times